aiida_vasp/calcs/vasp.py

```diff
@@ -69,24 +69,17 @@ def define(cls, spec):
         super(VaspCalculation, cls).define(spec)
         # Define the inputs.
         # options is passed automatically.
-        spec.input('parameters', valid_type=get_data_class('dict'), help='The VASP input parameters (INCAR).', required=False)
+        spec.input('parameters', valid_type=get_data_class('dict'), help='The VASP input parameters (INCAR).')
         spec.input('dynamics',
                    valid_type=get_data_class('dict'),
                    help='The VASP parameters related to ionic dynamics, e.g. flags to set the selective dynamics',
                    required=False)
-        spec.input('structure',
-                   valid_type=(get_data_class('structure'), get_data_class('cif')),
-                   help='The input structure (POSCAR).',
-                   required=False)
+        spec.input('structure', valid_type=(get_data_class('structure'), get_data_class('cif')), help='The input structure (POSCAR).')
 
         # Need namespace on this as it should also accept keys that are of `kind`. These are unknown
         # until execution.
-        spec.input_namespace('potential',
-                             valid_type=get_data_class('vasp.potcar'),
-                             help='The potentials (POTCAR).',
-                             dynamic=True,
-                             required=False)
-        spec.input('kpoints', valid_type=get_data_class('array.kpoints'), help='The kpoints to use (KPOINTS).', required=False)
+        spec.input_namespace('potential', valid_type=get_data_class('vasp.potcar'), help='The potentials (POTCAR).', dynamic=True)
+        spec.input('kpoints', valid_type=get_data_class('array.kpoints'), help='The kpoints to use (KPOINTS).')
```

Comment:
Here we block the functionality of running VASP with the INCAR flags for k-point meshes. A few people indeed uses that instead of the KPOINTS file. The provenance info of the k-point grid would then be determined by the code. Forcing a `KPOINTS` input would ensure that we control provenance. But as you know, if something happens with the symmetry routines which generate the IBZ, on which most of the calcs are being done, we anyway loose it.

Also, we should think a bit about what we will do when we do restarts and use remote folder for restarts. Then we might not need so many required inputs. I think the `False` on many of these was triggered by exactly this. The dynamic of the `potential` also opens for empty content. But I do not quite remember why it was done.

Reply:
Moving this to an issue.